### PR TITLE
ASRS spider uncommon random event

### PR DESCRIPTION
**Pull request by PhantomEpicness**
# About the pull request
New random event of ASRS spiders. Can vary from minor to extreme.
Uncommon (weight of 7)
Earliest start: 10 minutes
Requires 75 minimum players so deadpop shipsiders won't get screwed or the one guy playing req end up perma and forgotton

How it works: Picks a threat level percentage from 1 to 100. This is divided by 10 and then used as a multiplier for the base point amount (80) and rounded to the nearest 25 to "purchase" spider types in order of biggest to smallest. Then it adds 25 to this amount for good measure

So the theoretical maximum is 825 points

Big spider: 75 points
hunter spider: 50 points
Nurse spider + egg: 25 points
spiderling: Spawned in every tick

So the absolute MAXIMUM amount of spiders are:
8 big spiders
3 hunter spiders
3 nurse spiders
+ 14 spiderlings

It also has an RNG 5% chance for a "Matriarch" miniboss per spawn tick if it is over 500 points. Only one can be spawned.
The Matriarch spider can spit either neuro or acid, has 300 health and can lay 3 times the eggs. Robust, especially when with other spiders, but is a glass cannon as it still doesn't take much to kill it. 

# Explain why it's good for the game
Breaking up boringness and adding shipside fun

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>
### Video demonstrating a "stealth" infestation
https://user-images.githubusercontent.com/42422230/216804782-f97325e0-d0ec-47cd-9169-9e4705e207a1.mov
### screenshot demonstrating a medium level infestation
Screenshot_17

### Video demonstrating a maximum level infestation
_note: Message duplication is an earlier bug that is now fixed_
https://user-images.githubusercontent.com/42422230/216805352-6e9c1c83-809d-45be-9bbf-ab378c9fb6f9.mp4

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:TotalEpicness
add: Added an ASRS spider random event. This can scale from a full blown infestation on the ASRS that will alert to a couple surprise spiders. This is an uncommon event.
add: Slightly buffed the nurse spider's health from 40 to 75 as they are pretty much one hit.
add: Adds a Queen Spider subtype which is an ultra-rare spawn on the random event. She has 300 hp, spits acid and neuro, and spawns 3 times the eggs!
/:cl:

### CODE REVIEW

File: code/game/objects/effects/spiders.dm [hunk 1]
```diff
@@ -162,7 +162,7 @@
 	if(isturf(loc) && amount_grown > 0)
 		amount_grown += rand(0,2)
 		if(amount_grown >= 100)
-			var/spawn_type = pick(typesof(/mob/living/simple_animal/hostile/giant_spider))
+			var/spawn_type = pick(typesof(/mob/living/simple_animal/hostile/giant_spider,/mob/living/simple_animal/hostile/giant_spider/hunter,/mob/living/simple_animal/hostile/giant_spider/nurse))
```
Comment (harryob) [inline code/game/objects/effects/spiders.dm:165]:
```suggestion
			var/spawn_type = pick(typesof(/mob/living/simple_animal/hostile/giant_spider, /mob/living/simple_animal/hostile/giant_spider/hunter, /mob/living/simple_animal/hostile/giant_spider/nurse))
```

File: code/game/supplyshuttle.dm [hunk 2]
```diff
@@ -386,6 +386,8 @@ var/datum/controller/supply/supply_controller = new()
 
 	var/base_random_crate_interval = 10 //Every how many processing intervals do we get a random crates.
 
+	var/spider_infest_points = 0 // how many points a spider infestation has
```
Comment (harryob) [inline code/game/supplyshuttle.dm:389]:
```suggestion
	/// how many points a spider infestation has
	var/spider_infest_points = 0
```

File: code/game/supplyshuttle.dm [hunk 3]
```diff
@@ -565,7 +567,7 @@ var/datum/controller/supply/supply_controller = new()
 		dropped_limb.droplimb(FALSE, FALSE, "machinery")
 
 //Buyin
-/datum/controller/supply/proc/buy()
+/datum/controller/supply/proc/buy() //////////////////////////////////////////////////////////////
```
Comment (harryob) [inline code/game/supplyshuttle.dm:570]:
```suggestion
/datum/controller/supply/proc/buy()
```
why

File: code/game/supplyshuttle.dm [hunk 4]
```diff
@@ -1249,6 +1258,59 @@ var/datum/controller/supply/supply_controller = new()
...
+		if(elevator_turfs.density || elevator_turfs.contents?.len)
+			continue
+		clear_turfs |= elevator_turfs
+
+	/*
+	"Purchase" big ones, then small ones so it evens out
+
+	*/
+	while(total_points > 0)
+		log_debug("ASRS infestation event spawn tick")
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1285]:
```suggestion
```

File: code/game/supplyshuttle.dm [hunk 5]
```diff
@@ -1249,6 +1258,59 @@ var/datum/controller/supply/supply_controller = new()
...
 	return TRUE
 
+
+/datum/controller/supply/proc/init_infestation(base_factor,multiplier)
+	spider_infest_points = round(base_factor * multiplier/10,25)
+
+/datum/controller/supply/proc/spider_infest_asrs()
+	var/buy_big = TRUE 		// These exist to evenly spread the points so it isn't all big ones or hunters
+	var/buy_hunter = TRUE
+	var/queen_spawned = FALSE 	// So we don't spam Matriarch minibosses
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1268]:
```suggestion
	/// These exist to evenly spread the points so it isn't all big ones or hunters
	var/buy_big = TRUE
	var/buy_hunter = TRUE
	/// So we don't spam Matriarch minibosses
	var/queen_spawned = FALSE 	
```

File: code/modules/events/asrs_spiders.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/datum/round_event_control/asrs_spiders
+	name = "ASRS infestation"
+	typepath = /datum/round_event/asrs_spiders
+	weight = 7 							// probably could use some defines in the future - EVENT_UNLIKELY, EVENT_COMMON, etc.
+	earliest_start = 10 MINUTES 			// don't want it happening roundstart
+	min_players = 75 						// Probably not good for deadpop rounds where there might not even be dedicated req staff
+	max_occurrences = 2 					// any more and it'd get tiresome
```
Comment (harryob) [inline code/modules/events/asrs_spiders.dm:7]:
```suggestion
	weight = 7
	earliest_start = 10 MINUTES // don't want it happening roundstart
	min_players = 75 // Probably not good for deadpop rounds where there might not even be dedicated req staff
	max_occurrences = 2 // any more and it'd get tiresome
```

File: code/modules/events/asrs_spiders.dm [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+	alert_observers = TRUE
+	gamemode_blacklist = list("Whiskey Outpost", "Hive Wars")
+
+/datum/round_event/asrs_spiders
+	announce_when = 1
+	startWhen = 10
+	endWhen = 15
+	var/base_points = 80 // the base amount of points to be multiplied by the system
```
Comment (harryob) [inline code/modules/events/asrs_spiders.dm:15]:
```suggestion
	/// the base amount of points to be multiplied by the system
	var/base_points = 80
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 8]
```diff
@@ -34,33 +34,58 @@
...
+
+/mob/living/simple_animal/hostile/giant_spider/nurse/queen/AttackTarget()
+	..()
+	if(get_dist(src, target_mob) <= 5 && spit)
+		src.visible_message(SPAN_NOTICE("\the [src] spits at [target_mob]"))
+		var/obj/item/projectile/Pro = new(create_cause_data("Queen Spider Spit"))
+		var/spit_select = pick(/datum/ammo/xeno/acid,/datum/ammo/xeno/toxin/queen)
+		Pro.generate_bullet(new spit_select)
+		Pro.fire_at(target_mob,src,src, Pro.ammo.max_range, Pro.ammo.shell_speed, null)
+		Pro.permutated += src
```
Comment (harryob) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:84]:
```suggestion
		var/obj/item/projectile/spit_projectile = new(create_cause_data("Queen Spider Spit"))
		var/spit_select = pick(/datum/ammo/xeno/acid, /datum/ammo/xeno/toxin/queen)
		spit_projectile.generate_bullet(new spit_select)
		spit_projectile.fire_at(target_mob,src,src, spit_projectile.ammo.max_range, spit_projectile.ammo.shell_speed, null)
		spit_projectile.permutated += src
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 9]
```diff
@@ -34,33 +34,58 @@
...
+	maxHealth = 300
+	health = 300
+	isspiderqueen = TRUE
+	fed = 1
+	var/spit = TRUE
+
+/mob/living/simple_animal/hostile/giant_spider/nurse/queen/AttackTarget()
+	..()
+	if(get_dist(src, target_mob) <= 5 && spit)
+		src.visible_message(SPAN_NOTICE("\the [src] spits at [target_mob]"))
```
Comment (harryob) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:79]:
```suggestion
		src.visible_message(SPAN_NOTICE("[src] spits at [target_mob]"))
```
\the is implicit

File: ColonialMarinesALPHA.dme [hunk 10]
```diff
@@ -1538,6 +1538,7 @@
 #include "code\modules\emoji\emoji_parse.dm"
 #include "code\modules\events\_events.dm"
 #include "code\modules\events\comms_blackout.dm"
+#include "code\modules\events\asrs_spiders.dm"
```
Comment (harryob) [inline ColonialMarinesALPHA.dme:1541]: this is out of order and will fail linters

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
This is way too goofy and doesn't fit the setting at all imo

**Review by PhantomEpicness — COMMENTED** [review]
ok

**Comment by stanalbatross** [comment]
dnm applied for turther internal discussion

**Pull request closed without merging**